kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ApplicationOperation.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine
+
+import org.apache.kyuubi.config.KyuubiConf
+
+trait ApplicationOperation {
+
+  /**
+   * Step for initializing the instance.
+   */
+  def initialize(conf: KyuubiConf): Unit
+
+  /**
+   * Step to clean up the instance
+   */
+  def stop(): Unit
+
+  /**
+   * Called before other method to do a quick skip
+   *
+   * @param clusterManager the underlying cluster manager or just local instance
+   */
+  def isSupported(clusterManager: Option[String]): Boolean
```

Comment:
shall we pass engine type as a input parameter ?  for example: it's obvious that we can not kill trino with jps operation.

Reply:
we can kill trino via jps

Reply:
oh .. I mean yarn

Reply:
we can add this whenever we found it's necessary, for the current status, this just makes the PR larger and harder to review. WDYT?

Reply:
Besides， `clusterManager` for hive/trino/flink now are all None which bypassing for yarn operation.

Reply:
> we can add this whenever we found it's necessary

I'm fine with it if now here we do not require